### PR TITLE
EMPs Disable Cyborg Binary

### PR DESCRIPTION
**Pull request by Chakirski**
Fixes issue #4782 and adds functionality to disable robot components for a timed duration, however only the binary communications device has been modified in this at the moment.

Binary disable time is the same as the stun time.

### CODE REVIEW

File: code/modules/mob/living/silicon/robot/component.dm [hunk 1]
```diff
@@ -121,7 +125,13 @@
 
 /mob/living/silicon/robot/proc/is_component_functioning(module_name)
 	var/datum/robot_component/C = components[module_name]
-	return C && C.installed == 1 && C.toggled && C.is_powered()
+	return C && C.installed == 1 && C.toggled && C.is_powered() && C.not_disabled()
```
Comment (Fox-McCloud) [inline code/modules/mob/living/silicon/robot/component.dm]: You could probably just do `return C && C.installed == 1 && C.toggled && C.is_powered() && !C.component_disabled`

File: code/modules/mob/living/silicon/robot/component.dm [hunk 2]
```diff
@@ -121,7 +125,13 @@
...
+
+/mob/living/silicon/robot/proc/disable_component(module_name, emp_duration)
+	var/datum/robot_component/D = get_component(module_name)
+	D.component_disabled = 1
+	spawn (emp_duration)
+	D.component_disabled = 0
```
Comment (Fox-McCloud) [inline code/modules/mob/living/silicon/robot/component.dm]:
```
/mob/living/silicon/robot/proc/disable_component(module_name, emp_duration)
	var/datum/robot_component/D = get_component(module_name)
	D.component_disabled = 1
	spawn (emp_duration)
		D.component_disabled = 0
```

File: code/modules/mob/living/silicon/silicon.dm [hunk 3]
```diff
@@ -66,14 +66,16 @@
 /mob/living/silicon/drop_item()
 	return
 
-/mob/living/silicon/emp_act(severity)
+/mob/living/silicon/robot/emp_act(severity)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/silicon/silicon.dm]: this should be untouched; you should make a `/mob/living/silicon/robot/emp_act(severity)` and have it call to parent, then do different things based on severity.

File: code/modules/mob/living/silicon/silicon.dm [hunk 4]
```diff
@@ -66,14 +66,16 @@
 /mob/living/silicon/drop_item()
 	return
 
-/mob/living/silicon/emp_act(severity)
+/mob/living/silicon/robot/emp_act(severity)
 	switch(severity)
 		if(1)
 			src.take_organ_damage(20)
 			Stun(8)
+			src.disable_component("comms", 160)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/silicon/silicon.dm]: `src` is implied here, so you don't have to use it; just call the proc and that's it.

**Comment by FreeStylaLT** [comment]
..well then?

**Pull request closed without merging**